### PR TITLE
openssl 1.1.1n

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Actions:
1. Remove linux-generic32 from build.sh
2. Fix source/url with https

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
I wonder how conda-forge try to build v1.1.1n (https://github.com/conda-forge/openssl-feedstock/pull/83/files):
- To break circular dependency openssl -> clang -> ld64 -> sigtool -> openssl:
    `- {{ compiler('clang_bootstrap') }}  # [osx]`
- move `perl *` from `host` to `build` requirements

Should we use the same changes?

**Comment by chenghlee** [comment]
> * To break circular dependency openssl -> clang -> ld64 -> sigtool -> openssl:
>   `- {{ compiler('clang_bootstrap') }}  # [osx]`

I don't think we need to do this unless using just `{{ compiler('c') }}` causes build failures on our macOS workers.

> * move `perl *` from `host` to `build` requirements

I recommend we pick up this change, as long as the Windows issue described in the current comment does not re-occur.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
 source:
   fn: {{ name }}-{{ version }}.tar.gz
-  url: http://www.openssl.org/source/{{ name }}-{{ version }}.tar.gz
-  sha256: f89199be8b23ca45fc7cb9f1d8d3ee67312318286ad030f5316aca6462db6c96
+  url: https://www.openssl.org/source/{{ name }}-{{ version }}.tar.gz
+  sha256: 40dceb51a4f6a5275bde0e6bf20ef4b91bfc32ed57c0552e2e8e15463372b17a
   patches:
-    - 0001-Don-t-use-USE_BCRYPTGENRANDOM-for-VS-older-than-2015.patch
+    # backport https://github.com/openssl/openssl/pull/17902
+    - patches/gh17902.patch
```
Comment (chenghlee) [inline recipe/meta.yaml]: This backport seems to be from the OpenSSL 3.x branch, so I don't think we need to adopt it for our OpenSSL 1.1.1* build.

**Review by chenghlee — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: I removed it

**Comment by skupr-anaconda** [comment]
> > * move `perl *` from `host` to `build` requirements
> 
> I recommend we pick up this change, as long as the Windows issue described in the current comment does not re-occur.

I moved `perl` from `host` to `build` but it failed on win32 and win64

**Review by varlackc — COMMENTED** [review]
The package looks good, however the `home_url` need to use `https`

**Review by chenghlee — APPROVED** [review]

**Comment by skupr-anaconda** [comment]
> The package looks good, however the `home_url` need to use `https`

@varlackc Thank you. I fixed it

**Review by varlackc — APPROVED** [review]

**Review by chenghlee — APPROVED** [review]

**Review by tobijk — APPROVED** [review]
LGTM

**Comment by h-vetinari** [comment]
> I moved `perl` from `host` to `build` but it failed on win32 and win64

The key to that was to change `%LIBRARY_BIN%\perl` in bld.bat to just `perl`, see https://github.com/conda-forge/openssl-feedstock/commit/7ba0303aa55ace703d0665edd457454d6a1b8b28

Also, is there a publicly available feedstock for openssl 3.0.0? I'd like to sync this back with conda-forge, because I saw that Anaconda managed to get cryptography to run with openssl 3.0.0, and I've been struggling to get that off the ground.

**Pull request merged**